tests/Graphics/Implicit/Test/Instances.hs

```diff
@@ -235,7 +235,7 @@ isValid3 (CubeR _ (x0, y0, z0)) = (0 < x0) && (0 < y0) && (0 < z0)
 isValid3 (Cylinder _ r h) = r > 0 && h > 0
 isValid3 s =  -- Otherwise, make sure it has > 0 volume
   let (dx, dy, dz) = boxSize s
-   in notElem 0 [dx, dy, dz]
+   in all (> 0) [dx, dy, dz]
```

Comment:
pedantic note, should stuff thats ~ 1 ULP or so in every dim be legal??

Reply:
what's a ULP?

Reply:
"unit of least precision". though wikipedia has an equivalent but slightly different acronym https://en.wikipedia.org/wiki/Unit_in_the_last_place 

i probably mean "epsilon, the smallest normalized nonzero positive floating point"

Reply:
Tests pass, so I'm OK with it :)